sound/soc/sof/topology.c

```diff
@@ -2017,6 +2042,60 @@ static int sof_get_control_data(struct snd_soc_component *scomp,
 	return 0;
 }
 
+/*
+ * Sof mux/demux component
+ */
+static int sof_process_load_mux(struct snd_soc_component *scomp,
+				struct sof_ipc_comp_process *process,
+				struct snd_soc_tplg_private *private)
+{
+	struct sof_mux_config *mc = (struct sof_mux_config *)&process->data;
+	struct mux_stream_data *msd = (struct mux_stream_data *)
+		((u8 *)process->data + sizeof(struct sof_mux_config));
+	struct snd_soc_tplg_vendor_array *varray;
+	int pos = 0;
+	int ret;
+	int i;
+
+	/* parse mux config array */
+	ret = sof_parse_tokens(scomp, mc, mux_tokens,
+			       ARRAY_SIZE(mux_tokens), private->array,
+			       le32_to_cpu(private->size));
+	if (ret < 0) {
+		dev_err(scomp->dev, "error: parse mux tokens failed %d\n",
+			private->size);
+		return ret;
+	}
+
+	/* parse mux stream config arrays */
+	for (i = 0; i < mc->num_streams; i++) {
+		varray = (struct snd_soc_tplg_vendor_array *)
+			((u8 *)private->array  + pos);
+		ret = sof_parse_tokens(scomp, msd, mux_stream_tokens,
+				       ARRAY_SIZE(mux_stream_tokens),
+				       varray,
+				       le32_to_cpu(private->size) - pos);
+		if (ret < 0) {
+			dev_err(scomp->dev, "error: mux stream tokens %d\n",
+				private->size);
+			return ret;
+		}
+
+		/* track position for multiple same type arrays */
+		pos += ret;
+		msd++;
+	}
+
+	/* get the channel byte masks after stream arrays */
+	msd -= mc->num_streams;
+	for (i = 0; i < mc->num_streams; i++) {
```

Comment:
ditto

Reply:
ok, so the added things we're here, not above.